### ISSUE TITLE
OracleDataSource cannot be used with the Datasource auto-configuration

### ISSUE THREAD

**Issue by chtinnes** [spring-projects/spring-boot#6027]
I got some problems using a OracleDataSource with Spring Boot Autoconfiguration.
In my properties file, I set the spring.datasource.type to oracle.jdbc.pool.OracleDataSource.
Somehow, the DataSourceBuilder is not setting the username/name property for the OracleDataSource. 
As a workaround, one can set password and username in the connection string.

**Comment by snicoll** [comment]
We're not aware of any problem of the kind. Can you show your configuration? What Spring Boot version are you using?

**Comment by chtinnes** [comment]
We are using version 1.3.0.RELEASE. 

spring.datasource.url=jdbc:oracle:thin:server:xe
spring.jpa.database-platform=org.hibernate.dialect.Oracle10gDialect
spring.datasource.username=usename
spring.datasource.password=password
spring.datasource.driver-class-name=oracle.jdbc.OracleDriver
spring.datasource.type=oracle.jdbc.pool.OracleDataSource

What configuration do you mean?

**Comment by snicoll** [comment]
`OracleDataSource` does not have any setter and we're relying on that to transmit the state of the builder to the target data source. The Oracle implementation forces us  to use a special constructor that has the url + a `java.util.Properties` with the settings.

**Comment by snicoll** [comment]
`spring.datasource.type=oracle.jdbc.pool.OracleDataSource`

@chtinnes we don't support that datasource, check [the documentation](http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-connect-to-production-database) for the list of supported data-sources. So it's not a bug report from that perspective.

**Comment by chtinnes** [comment]
We want to use the same datasource for all db accesses. Amog others, we use AQJms. I think for this need a DataSource like OracleDataSource. Or is there any other way to do this?

**Comment by snicoll** [comment]
You can but the auto-configuration supports 4 data source implementations (as explained in the doc) and not this one so if you want to use it, no problem but you'll have to configure it yourself, something like

```
@Configuration
@EnableConfigurationProperties(DataSourceProperties.class)
public class CustomConfiguration {

	@Bean
	public DataSource dataSource(DataSourceProperties properties) throws SQLException {
		OracleDataSource dataSource = new OracleDataSource();
		dataSource.setUser(properties.getUsername());
		dataSource.setPassword(properties.getPassword());
		dataSource.setURL(properties.getUrl());
		// more settings
		return dataSource;
	}
}
```

You can still use the `spring.datasource.*` and they will bound to `DataSourceProperties`. 

@philwebb I am not sure I like the idea of adding support for this connection pool implementation. What do you think?

**Comment by shakuzen** [comment]
I believe whether to support auto-configuration of `OracleDataSource` is still something that could be decided in this issue, but just to add context to the topic of using AQ JMS (I have no experience with it), [this seems to indicate](http://docs.spring.io/spring-data/jdbc/docs/current/reference/html/orcl.streamsaq.html) that you can use other `DataSource` implementations with AQ JMS. It's not required that it is `OracleDataSource`, I would suppose (the example shows Commons DBCP). Given that, if @chtinnes does not have some other requirement to use `OracleDataSource`, it seems the goals could be accomplished with one of the auto-configuration supported DBCP implementations.

**Comment by wilkinsona** [comment]
I don't think we have any choice here. `OracleDataSource` isn't available in any public Maven repository and, as far as I can tell, the terms of its [license](http://www.oracle.com/technetwork/licenses/distribution-license-152002.html) prohibit us from adding it to one of our own.

**Comment by snicoll** [comment]
@chtinnes you may want to configure your pool using `OracleConnectionPoolDataSource` instead. See #6124 that we're going to fix in a future release.

**Comment by snicoll** [comment]
Scratch that: I can't find it back right now but I managed to find an `OracleDataSource` javadoc somewhere that did not expose any setter but a constructor that takes a `Properties` instance only.

**Comment by snicoll** [comment]
Fixed in 6dde498
